x-pack/plugins/apm/server/lib/connections/get_connection_stats/get_stats.ts

```diff
@@ -54,7 +55,142 @@ export const getStats = async ({
     offset,
   });
 
-  const response = await apmEventClient.search('get_connection_stats', {
+  const response = await getAllConnectionStats({
+    apmEventClient,
+    startWithOffset,
+    endWithOffset,
+    filter,
+    numBuckets,
+  });
+
+  return (
+    response.aggregations?.connections.buckets.map((bucket) => {
+      const sample = bucket.sample.top[0].metrics;
+      const serviceName = bucket.key.serviceName as string;
+      const dependencyName = bucket.key.dependencyName as string;
+
+      return {
+        from: {
+          id: objectHash({ serviceName }),
+          serviceName,
+          environment: (sample[SERVICE_ENVIRONMENT] ||
+            ENVIRONMENT_NOT_DEFINED.value) as string,
+          agentName: sample[AGENT_NAME] as AgentName,
+          type: NodeType.service as const,
```

Comment:
Q Why do you need the const here the type looks correct without it or I am missing something: 
<img width="213" alt="image" src="https://github.com/elastic/kibana/assets/14139027/322e529d-0c0f-4920-86da-abd7770dd6b8">

Reply:
I don't think it's needed here, to be honest. However, in other places, it's useful because you tell typescript to treat the type as literal rather than general, like the ones in the `getConnectionStats` function. With that, you can easily access the `aggregations.connections` attributes.